### PR TITLE
Support customizing standard accessibility action hints on Android.

### LINKED ISSUE #14402: Make "double-tap to activate" customizable on Android
Currently, whenever something is tap-able, Android gives the hint "double-tap to activate" and Flutter doesn't provide a way to override it. If you provide a custom hint via the Semantics widget, both hints (custom and default) are read out because we just add the custom hint to the regular text of the node (on iOS, the custom hint replaces the default hint as expected).

Android doesn't really provide a way to replace the default hint. Instead, starting with API 21 it can be customized via custom Accessibility Actions [1]. The description string given there will replace the "\<action\>" in "double-tap to \<action\>".

We should find a way to support that in Flutter.

[1] https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo.AccessibilityAction.html#AccessibilityNodeInfo.AccessibilityAction(int, java.lang.CharSequence)

### PR DESCRIPTION
**Pull request by jonahwilliams**
requires https://github.com/flutter/engine/pull/5823/files

Fixes https://github.com/flutter/flutter/issues/14402

### CODE REVIEW

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 1]
```diff
@@ -1437,7 +1536,7 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
 
     assert(
       !_canPerformAction(SemanticsAction.increase) || (_value == '') == (_increasedValue == ''),
-      'A SemanticsNode with action "increase" needs to be annotated with either both "value" and "increasedValue" or neither',
+      'A SemanticsNode with actixon "increase" needs to be annotated with either both "value" and "increasedValue" or neither',
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart]: typo

**Review by goderbauer — COMMENTED** [review]
Can you add some tests?

**Comment by jonahwilliams** [comment]
I have https://github.com/flutter/flutter/pull/20095 pending to add the basic customActions support to the matcher - I can fold that into this PR and do the test support as well if that makes sense?

**Comment by goderbauer** [comment]
I think that makes sense.

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/semantics/semantics.dart]: Fixed

**Comment by jonahwilliams** [comment]
Tests added for matcher and semantics widget.  I'll expand test coverage a bit when I actually take advantage of these fields for tooltips, chips, and expansion panels.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 2]
```diff
@@ -98,20 +98,44 @@ class CustomSemanticsAction {
...
-      assert(label != '');
+      assert(label != ''),
+      hint = null,
+      action = null;
 
-  /// The user readable name of this custom accessibility action.
+  /// Creates a new [CustomSemanticsAction] that shadows a standard semantics action.
+  ///
+  /// The [hint] must not be null or the empty string.
+  const CustomSemanticsAction.standard({@required this.hint, @required this.action})
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter/lib/src/semantics/semantics.dart]: Done

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 3]
```diff
@@ -407,6 +430,64 @@ class _SemanticsDiagnosticableNode extends DiagnosticableNode<SemanticsNode> {
   }
 }
 
+/// Provides hint values which override the default hints on supported
+/// platforms.
+///
+/// On iOS, these are always ignored.
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter/lib/src/semantics/semantics.dart]: I don't have guidance for this yet - it _seems_ like you would only want a hint on iOS and hintOverrides on android.

**Pull request merged**